File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.LinkedList;

public class Main {	
	public static void main(String[] args) throws Exception {
        
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));		
		
		int N =  Integer.parseInt(br.readLine());
		LinkedList<Integer> qu = new LinkedList<Integer>();
		
		for(int i=0; i<N; i++) {
			String[] input = br.readLine().split(" ");
			int output = -1;
			
			switch(input[0]) {
			
				case "push":
					qu.offer(Integer.parseInt(input[1]));
					break;
					
				case "pop":
					output = -1;
					if(!qu.isEmpty()) {
						output = qu.poll();
					}
					bw.write(output + "\n");
					break;
					
				case "size":
					output = qu.size();
					bw.write(output + "\n");
					break;
					
				case "empty":
					output = 1;
					if(!qu.isEmpty()) {
						output = 0;
					}
					bw.write(output + "\n");
					break;
					
				case "front":
					output = -1;
					if(!qu.isEmpty()) {
						output = qu.peek(); 
					}
					bw.write(output + "\n");
					break;
				case "back":
					output = -1;
					if(!qu.isEmpty()) {
						output = qu.peekLast();
					}
					bw.write(output + "\n");
					break;		
			}
		}				
	    bw.flush();
		br.close();
		bw.close();	
	}	
}

File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

cnt = int(sys.stdin.readline())
queue = deque([])

for i in range(cnt):
    com = sys.stdin.readline().split()
    if com[0] == 'push':
        queue.append(com[1])
    elif com[0] == 'pop':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.popleft())
    elif com[0] == 'size':
        print(len(queue))
    elif com[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    elif com[0] == 'front':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])
    elif com[0] == 'back':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])


# 시간 초과

# import sys
# cnt = int(sys.stdin.readline())
# queue = []

# for i in range(cnt):
#     command = sys.stdin.readline().split()
#     if command[0] == 'push':
#         queue.append(command[1])
#     elif command[0] == 'pop':
#         if len(queue) == 0:
#             print(-1)
#         else:
#             print(queue.pop(0))  
#     elif command[0] == 'size':
#         print(len(queue))
#     elif command[0] == 'empty':
#         if len(queue) == 0:
#             print(1)
#         else:
#             print(0)
#     elif command[0] == 'front':
#         if len(queue) == 0:
#             print(-1)
#         else:
#             print(queue[0])
#     elif command[0] == 'back':
#         if len(queue) == 0:
#             print(-1)
#         else:
#             print(queue[-1])